lib/custom_cache_file_store.rb

```diff
@@ -0,0 +1,28 @@
+class CustomCacheFileStore < ActiveSupport::Cache::MemoryStore
+
+  def initialize(*args)
+    super(*args)
+    @cache_keys = {}
+  end
+
+  def write(*args)
+    super
+    @cache_keys[ args[0] ] = true
+  end
+
+  def delete(*args)
+    super
+    @cache_keys.delete(args[0])
+  end
+
+  def keys
+    @cache_keys.keys
+  end
```

Comment:
Why can't you overwrite only `#clear` and left the rest of methods in the super class?

Reply:
Because the superclass does not provide a method to get the list of the existing keys, that's why I'm recording them in `@cache_keys`

Reply:
OK, now I get it.